### DISCUSSION TITLE
Read-Write property with Value conversion

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3652] [Questions]
Hello fellow Csla developers,
We have a small app where we have a couple of radio buttons for something like gender, one is for 'male' and the other for 'female'. (not exactly our use-case but just using it as an example). The values for gender are coming from a lookup table with rows as 'Male' and 'Female'.

One suggestion was to use two csla read-write properties
   1) IsMale
   2) IsFemale
but I have never tried my versions of SetPropertyConvert and GetPropertyConvert.

I guess my question is about writing our custom SetPropertyConvert and GetPropertyConvert

Kind Regards

**Comment by rockfordlhotka** [comment]
Sometimes the answer is to use a viewmodel (or equivalent UI framework things like value converters) to do this work. The business domain types should reflect the user workflow and thus are _close to_ the needs of the UI, but sometimes there does need to be some presentation layer code to make things work for a specific UI design requirement that isn't really a business requirement.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka  Thank you Sir. 

Somewhat related to the above question, if we have a string property that is used to interact with the private SmartDate instance's Text property, where should we handle the exception thrown by SetPropertyConvert ?

Kind Regards
